### PR TITLE
Modular ALSA pipeline for Volumio

### PR DESCRIPTION
**Pull request by timothyjward**
This contribution tidies up the work I originally created to let plugins provide a contribution to the Volumio ALSA configuration. The support can be enabled by setting the relevant environment variable `MODULAR_ALSA_PIPELINE=true`, and is disabled by default. This allows the feature to be added and tested by interested users, without impacting users and plugins that aren't ready for the ALSA changes.

It is recommended that when setting `MODULAR_ALSA_PIPELINE=true` that `WRITE_MPD_CONFIGURATION_ON_STARTUP=true` is also set, otherwise MPD may use the wrong configuration until it is next configured.

Replaces PR #1988

### CODE REVIEW

File: app/plugins/music_service/mpd/index.js [hunk 1]
```diff
@@ -955,18 +957,14 @@ ControllerMpd.prototype.createMPDFile = function (callback) {
...
-        if (outdev.indexOf(',') >= 0) {
-          mixerdev = 'hw:' + outdev;
-          outdev = 'hw:' + outdev;
+      if (!softvolume) {
+        if (realDev.indexOf(',') >= 0) {
+          mixerdev = 'hw:' + realDev;
         } else {
-          mixerdev = 'hw:' + outdev;
-          outdev = 'hw:' + outdev + ',0';
+          mixerdev = 'hw:' + realDev;
```
Comment (ashthespy) [inline app/plugins/music_service/mpd/index.js]: Not sure if this is an oversight, but isn't this the `else` branch here the same as in the `if` branch?

**Comment by balbuze** [comment]
sorry but I had exported both env variable, switched to the branch but it seems to be not enabled. Help welcome ;-)

**Comment by timothyjward** [comment]
> sorry but I had exported both env variable, switched to the branch but it seems to be not enabled. Help welcome ;-)

Did you try changing `/volumio/.env`? https://github.com/volumio/Volumio2/blob/f7c9313d514c4cde2e012fc9c959de8440224872/.env#L15

**Comment by balbuze** [comment]
ok, it works now! ;-)

**Comment by balbuze** [comment]
hi! Any news about this PR?

**Comment by timothyjward** [comment]
Closing as #2040 is now in the buster development stream.

**Pull request closed without merging**